_rules/SC4-1-2-aria-attr-valid.md

```diff
@@ -0,0 +1,118 @@
+---
+name: Validity of ARIA attribute
+
+description: |
+ This rule checks that the `aria-` attribute specified is defined in ARIA 1.1.
+
+success_criterion:
+- 4.1.2
+
+test_aspects:
+- DOM Tree
+
+authors:
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+Any attribute that starts with `aria-`.
+
+### Expectation
+
+Each target attribute which starts with `aria-` is defined in [WAI ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/).
+
+## Assumptions
+
+`aria-` attributes are used to provide assistive technologies with accessibility information. This rule does not cater to any custom `aria-` attribute that does not fall under the [WAI ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/) specification.
```

Comment:
Not sure if we need the first sentence. But it does make it easier to understand the second.

Reply:
I agree with removing the first sentence.

Reply:
Quoting @kasperisager from https://github.com/auto-wcag/auto-wcag/pull/233: 
> "If someone happens to be using aria-* attributes for something other than what is defined by WAI-ARIA, the rule would still be correct; whoever is doing such a thing should stop as they're messing with well-defined semantics of the web. Otherwise, every rule should include assumptions like "we assume that the author is using <img> elements for images, not for expressing custom state used in scripts". That just seems silly"